packages/core/src/App/Components/Routes/route-with-sub-routes.jsx

```diff
@@ -15,11 +15,14 @@ import { connect } from 'Stores/connect';
 
 const RouteWithSubRoutes = route => {
     const validateRoute = pathname => {
-        if (pathname.startsWith('/cashier') && !pathname.startsWith('/cashier/p2p/') && !!route.routes) {
-            return route.path === pathname || !!route?.routes.find(r => pathname === r.path);
-        } else if (pathname.startsWith('/cashier/p2p/') && !!route.routes) {
-            const cashier_subroutes = route?.routes.find(r => r.path === '/cashier/p2p');
-            const p2p_subroutes = cashier_subroutes?.routes.find(r => pathname === r.path);
+        if (pathname.startsWith('/cashier') && !pathname.includes('p2p') && !!route.routes) {
+            return route.path === pathname || !!route?.routes.find(({ path }) => pathname === path);
+        } else if (pathname.includes('p2p') && !!route.routes) {
+            const cashier_subroutes = route?.routes.find(({ path }) => path === '/cashier/p2p');
+            const p2p_subroutes =
+                pathname === '/cashier/p2p'
+                    ? routes.p2p_buy_sell
+                    : cashier_subroutes?.routes.find(({ path }) => pathname === path);
```

Comment:
since the pathname /cashier/p2p doesn't exist in the p2p subroutes, we instead default the path to buy-sell page or else it'll just return undefined and wont route to p2p